apis/cpi/v1alpha1/cpiconfig_types.go

```diff
@@ -0,0 +1,205 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// CPIConfigSpec defines the desired state of CPIConfig
+type CPIConfigSpec struct {
+	VSphereCPI VSphereCPI `json:"vsphereCPI"`
+}
+
+type NsxTRoute struct {
+	// NSX-T T0/T1 logical router path
+	// +kubebuilder:validation:Optional
+	RouterPath string `json:"routerPath"`
+
+	// Cluster CIDR
+	// +kubebuilder:validation:Optional
+	ClusterCidr string `json:"clusterCidr"`
+}
+
+type NsxT struct {
+	// A flag that enables pod routing
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=false
+	PodRoutingEnabled bool      `json:"podRoutingEnabled"`
+	Routes            NsxTRoute `json:"routes"`
+
+	// The username used to access NSX-T
+	// +kubebuilder:validation:Optional
+	Username string `json:"username"`
```

Comment:
Validation could be added to `nsxt.username` and `nsxt.password` if nsxt is enabled when implementing the controller.

Reply:
sure, drop the validation here for now